### PR TITLE
Table docs: updates to further clarify alignment

### PR DESCRIPTION
**Pull request by MelSumner**
### :pushpin: Summary

If merged, this PR updates the prose for vertical alignment and text alignment for tables, and makes other small cleanup fixes.

Preview link: https://hds-website-git-melsumner-hds-1586-alignment-hashicorp.vercel.app/components/table?tab=code

### :hammer_and_wrench: Detailed description

- added "how to use" examples for alignment
    - vertical alignment
    - vertical alignment with more than just text content
    - alignment (text). 
- added info box for vertical alignment with links to MDN
- added "how to use" example for density
- removed pre-adoption example
- removed intro paragraph
- added heavily condensed intro paragraph into an info box
- removed commented file names from examples
- updated API prose for `valign` for additional clarity
- updated API prose for `align` for additional clarity
- updated some example code line-breaks for WCAG conformance

### :camera_flash: Screenshots

CleanShot 2023-03-13 at 09 45 57

CleanShot 2023-03-13 at 10 24 52

CleanShot 2023-03-13 at 10 25 20

CleanShot 2023-03-13 at 10 25 37

### :link: External links

Jira ticket: [HDS-1586](https://hashicorp.atlassian.net/browse/HDS-1586)
Figma file: [if it applies]

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1586]: https://hashicorp.atlassian.net/browse/HDS-1586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/components/table/partials/code/how-to-use.md [hunk 1]
```diff
@@ -216,22 +213,128 @@ get customSortingMethodForStatus() {
...
+
+To create a column that has right-aligned content, set `@align` to `right` on both the column's header and cell.
+
+```handlebars{data-execute=false}
+<Hds::Table
+  @model={{this.model.data}}
+  @columns={{array
+    (hash key="artist" label="Artist" isSortable=true)
+    (hash key="album" label="Album" isSortable=true)
+    (hash key="year" label="Release Year" align=right)
```
Comment (MelSumner) [inline website/docs/components/table/partials/code/how-to-use.md:318]:
```suggestion
    (hash key="year" label="Release Year" align="right")
```

File: website/docs/components/table/partials/code/how-to-use.md [hunk 2]
```diff
@@ -141,7 +140,6 @@ You can optionally indicate that a specific column should be pre-sorted by addin
 You can optionally also indicate that the column defined in `@sortBy` should be pre-sorted in a descending order, rather than the default ascending order by passing in `@sortOrder='desc'`.
```
Comment (heatherlarsen) [inline website/docs/components/table/partials/code/how-to-use.md:140]:
```suggestion
By default, the sort order is set to ascending. To indicate that the column defined in `@sortBy` should be pre-sorted in descending order, pass in `@sortOrder='desc'`.
```

File: website/docs/components/table/partials/code/how-to-use.md [hunk 3]
```diff
@@ -167,7 +165,6 @@ You can optionally also indicate that the column defined in `@sortBy` should be
 To implement a custom sort callback on a columns, (1) add a custom function as the value for `sortingFunction` in the column hash, and (2) include a custom `onSort` action in your table invocation to track the sorting order and use it in the custom sorting function. This is useful for cases where the key might not be A-Z or 0-9 sortable by default, i.e., status, and you’re otherwise unable to influence the shape of the data in the model. Here is an example, code truncated for clarity:
```
Comment (heatherlarsen) [inline website/docs/components/table/partials/code/how-to-use.md:165]:
```suggestion
To implement a custom sort callback on a column:

1. add a custom function as the value for `sortingFunction` in the column hash,
2. include a custom `onSort` action in your table invocation to track the sorting order and use it in the custom sorting function. 

This is useful for cases where the key might not be A-Z or 0-9 sortable by default, e.g., status, and you’re otherwise unable to influence the shape of the data in the model. 

_The code has been truncated for clarity._
```

File: website/docs/components/table/partials/code/how-to-use.md [hunk 4]
```diff
@@ -216,22 +213,128 @@ get customSortingMethodForStatus() {
...
+
+#### Vertical alignment
+
+To indicate that the table's content should have a middle vertical-align, use `@valign` in the table's invocation.
+
+!!! Info
+
+Note that vertical-align only applies to inline, inline-block and table-cell elements: you can't use it to vertically align block-level elements ([see MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align)).
+
+If you have more than just text content in the table cell, you'll want to wrap that content in a flex box and style accordingly.
```
Comment (heatherlarsen) [inline website/docs/components/table/partials/code/how-to-use.md:258]: This content feels more related to the heading and example below than it does to the informational message above. What about moving it under the heading "Vertical alignment with additional cell content"?

**Review by heatherlarsen — COMMENTED** [review]
I may have missed the wagon on the review of the docs, so I spotted a few things that might benefit from iteration. They may be outside the scope of this PR, so feel free to ignore them and tackle them in a follow-up PR as necessary.

File: website/docs/components/table/partials/code/component-api.md [hunk 5]
```diff
@@ -53,7 +53,7 @@ The Table component itself is where most of the options will be applied. However
     If set, determines the density (height) of the table body’s rows.
   </C.Property>
   <C.Property @name="valign" @type="enum" @values={{array "top" "middle" }} @default="top">
-    If set, determines the vertical alignment for cell (`td`) content in a table. Does not apply to table headers (`th`).
+    If set, determines the vertical alignment for content in a table. Does not apply to table headers (`th`). See [MDN on vertical-align](https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align) for more details.
```
Comment (jorytindall) [inline website/docs/components/table/partials/code/component-api.md:56]:
```suggestion
    Determines the vertical alignment for content in a table. Does not apply to table headers (`th`). See [MDN on vertical-align](https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align) for more details.
```

Should this use the same language as the other properties?

File: website/docs/components/table/partials/code/how-to-use.md (hunk 4 above)
Comment (MelSumner) [reply website/docs/components/table/partials/code/how-to-use.md:265]: The goal was to explain why vertical align alone won't work, and what the consumer should do instead. WDYT?

File: website/docs/components/table/partials/code/how-to-use.md [hunk 6]
```diff
@@ -1,11 +1,14 @@
-This component takes advantage of the `sort-by` helper provided in [ember-composable-helpers](https://github.com/DockYard/ember-composable-helpers). While some of the examples provided on our documentation use some of the other helpers provided in this addon, they are not required to use Helios. Read the [configuration information](https://github.com/DockYard/ember-composable-helpers#configuration) provided by the addon if you wish to customize which helpers are included in your own app.
-
 ## How to use this component
 
+!!! Info
+
+This component takes advantage of the `sort-by` helper provided by [ember-composable-helpers](https://github.com/DockYard/ember-composable-helpers).
+
+!!!
```
[the remark this answers is not in the document]
Comment (MelSumner) [reply website/docs/components/table/partials/code/how-to-use.md:7]: I want this information up-front for a few reasons; one is that many apps are already this addon in their applications. So the idea is, this thing is probably already familiar to you. But there was other feedback that the previous iteration felt too wordy, so I tried to strip it down to the bare minimum. Does that help clarify? WDYT?

[the remark this answers is not in the document]
Comment (heatherlarsen) [reply website/docs/components/table/partials/code/how-to-use.md:7]: I like the stripped-down version. ✨ I'm more curious about its placement in the overall hierarchy of the docs because it's only needed for sortable tables, correct? If so, it feels like there's a more natural fit for this content within the section on sortable tables.

File: website/docs/components/table/partials/code/how-to-use.md (hunk 4 above)
Comment (heatherlarsen) [reply website/docs/components/table/partials/code/how-to-use.md:265]: Right, I understand that. What I'm suggesting is that the content within the info banner feels more related to the code example under the heading "Vertical alignment with additional cell content" than where it's placed now.

It feels like there's opportunity to move it here for a more natural flow of content:
Screenshot 2023-03-14 at 1 15 16 PM

File: website/docs/components/table/partials/code/how-to-use.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (didoo) [reply website/docs/components/table/partials/code/how-to-use.md:7]: @MelSumner not a blocker, but I strongly agree with @heatherlarsen: this is too prominent, and I think should be moved under the sortable tables examples where is more in context

**Review by didoo — APPROVED** [review]
👍

File: website/docs/components/table/partials/code/how-to-use.md [hunk 7]
```diff
@@ -83,16 +82,17 @@ You can insert your own content into the and `:body` block and the component wil
 **Important**
 
 For clarity, there are a couple of important points to note here:
+
 - provide a `@columns` argument (see [Component API](#component-api) for details about its shape)
 - use the `.data` key to access the `@model` record content (it’s yielded as `data`)
+
 !!!
```
[the remark this answers is not in the document]
Comment (MelSumner) [reply website/docs/components/table/partials/code/how-to-use.md:83]: Let's save this one for a separate PR, since it's not a change introduced by this PR and is out of scope for the associated JIRA ticket(s).

File: website/docs/components/table/partials/code/how-to-use.md (hunk 4 above)
Comment (MelSumner) [reply website/docs/components/table/partials/code/how-to-use.md]: I see what you mean. I've updated it per your suggestion.
